go/src/koding/kites/kloud/provider/google/stack.go

```diff
@@ -254,6 +257,10 @@ func (s *Stack) ApplyTemplate(c *stack.Credential) (*stack.Template, error) {
 
 	t.Resource["google_compute_instance"] = resource.GCInstance
 
+	t.Variable["koding_stack_id"] = map[string]interface{}{
```

Comment:
This is the very same problem which is not solved for AWS and other providers. We just did not hit the problem so far, so my take is to not half-solve-it. Just wait when it becomes a problem, and then design full solution.

I know it's unrelated, but I'd want to solve it by adding concept of koding stacks (generic across providers, right now each stack handles things like `debug`, `alwaysOn` on each own) and solve [this one at once](https://github.com/koding/koding/issues/9136). We're just not there yet, and I would want to restrain from extending the templates until needed.

You add here `koding_stack_id` and next thing what will happen is user asking how he could use it in AWS.